docs/table_design/expression_partitioning.md

```diff
@@ -0,0 +1,256 @@
+);
+```
+
+Example 3: Suppose you frequently query data by week.  You can use the partition expression `time_slice()` and set the partition column as `event_day` and the partition granularity to seven days at table creation. Data of one week is stored in one partition and partition pruning can be used to significantly improve query efficiency.
+
+```SQL
+CREATE TABLE site_access(
+    event_day DATETIME NOT NULL,
+    site_id INT DEFAULT '10',
+    city_code VARCHAR(100),
+    user_name VARCHAR(32) DEFAULT '',
+    pv BIGINT DEFAULT '0'
+)
+DUPLICATE KEY(event_day, site_id, city_code, user_name)
+PARTITION BY time_slice(event_day, INTERVAL 7 day)
+DISTRIBUTED BY HASH(event_day, site_id)
+```
+
+## Partitioning based on the column expression (since v3.1)
+
+If you frequently query and manage clearly categorized data based on specific types, you only need tospecify the column representing the type as the partition column. StarRocks will automatically create partitions based on the partition column values of the loaded data.
+
+However, in some special scenarios, such as  when the table contains a column `city`, and you frequently query and manage data based on countries and cities. You must use [list partitioning](./list_partitioning.md) to store data of multiple cities within the same country in one partition.
+
+### Syntax
+
+```sql
+PARTITION BY expression
+...
+[ PROPERTIES("partition_live_number" = "xxx") ]
+
+expression ::=
+    ( partition_columns )
+    
+partition_columns ::=
+    <column>, [ <column> [,...] ]
+```
+
+### Parameters
+
+| **Parameters**          | **Required** | **Description**                                                  |
+| ----------------------- | -------- | ------------------------------------------------------------ |
+| `partition_columns`     | YES      | The names of partition columns.<br> <ul><li>The partition column values can be string (BINARY not supported), date or datetime, integer, and boolean values. The partition column allows `NULL` values.</li><li> Each partition can only contain data with the same value for a partition column. To include data with different values in a partition column in a partition, see [List partitioning](./list_partitioning.md).</li></ul> |
+| `partition_live_number` | YES      | The number of partitions to be retained. Compare the values of partition columns among partitions, and periodically delete partitions with smaller values while retaining those with larger values.<br>StarRocks schedules tasks to manage the number of partitions, and the scheduling interval can be configured through the FE dynamic parameter `dynamic_partition_check_interval_seconds`, which defaults to 600 seconds (10 minutes).<br>**NOTE**<br>If the values in the partition column are strings, StarRocks compare the lexicographical order of the partition names, and periodically retains the partitions that come earlier while deleting the partitions that come later. |
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/897f35d6-a3eb-4a15-b781-664a83a2d639)
![image](https://github.com/StarRocks/starrocks/assets/57167462/975d2ff8-a305-4dce-a19c-3493ecb162c6)
Pls confirm if the parameter is required or not.